dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/DecodeableRpcResult.java

```diff
@@ -119,28 +121,38 @@ public void decode() throws Exception {
         }
     }
 
-    private void setResponseResult(ObjectInput in, boolean hasValue, boolean hasException, boolean hasAttachments) throws IOException {
+    private void handleValue(ObjectInput in) throws IOException {
         try {
-            if (hasValue) {
-                Type[] returnType = RpcUtils.getReturnTypes(invocation);
-                setValue(returnType == null || returnType.length == 0 ? in.readObject() :
-                        (returnType.length == 1 ? in.readObject((Class<?>) returnType[0])
-                                : in.readObject((Class<?>) returnType[0], returnType[1])));
-            }
-            if (hasException) {
-                Object obj = in.readObject();
-                if (obj instanceof Throwable == false) {
-                    throw new IOException("Response data error, expect Throwable, but get " + obj);
-                }
-                setException((Throwable) obj);
-            }
-            if (hasAttachments) {
-                setAttachments((Map<String, String>) in.readObject(Map.class));
+            Type[] returnType = RpcUtils.getReturnTypes(invocation);
+            setValue(returnType == null || returnType.length == 0 ? in.readObject() :
```

Comment:
can be refactore to a method 

```
private Object getInvocationReturnType() {
  if(returnType == null || returnType.length == 0) {
      return in.readObject();
  }
  if( returnType.length ==1 ) {
     return in.readObject((Class<?>) returnType[0];
  }
   return in.readObject((Class<?>) returnType[0], returnType[1]);
}
```
What do you say?
As this code I have done without editor so there might some braces or syntactical error so plz excuse me for that.

Reply:
agree.